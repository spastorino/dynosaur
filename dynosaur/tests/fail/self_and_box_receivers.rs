use std::rc::Rc;

#[dynosaur::dynosaur(DynAll = dyn(box) All)]
trait All {
    fn ref_mut(&mut self);
    fn ref_(&self) -> impl Send;
    fn owned(self) -> impl Send;
    fn self_box(self: Box<Self>) -> impl Send;
    fn self_rc(self: Rc<Self>) -> impl Send;
}

fn main() {}
