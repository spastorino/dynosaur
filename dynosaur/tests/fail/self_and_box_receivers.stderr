error: By value Self and Box<Self> are unsupported
 --> tests/fail/self_and_box_receivers.rs:2:1
  |
2 | / trait All {
3 | |     fn ref_mut(&mut self);
4 | |     fn ref_(&self) -> impl Send;
5 | |     fn owned(self) -> impl Send;
6 | |     fn self_box(self: Box<Self>) -> impl Send;
7 | | }
  | |_^

error: aborting due to 1 previous error

